main.go

```diff
@@ -179,8 +192,17 @@ func (d *Downloader) DownloadWithContext(ctx context.Context, urls ...string) <-
 				s.Error = err
 				return
 			}
+			ds, err := d.getContentSizeHeader(ctx, u)
+			if err != nil {
+				s.Error = err
+				return
+			}
 			s.DownloadedFileBytes = uint64(len(b))
-			s.FileSizeBytes = uint64(len(b))
+			if ds == 0 {
```

Comment:
From your practical experience, do we have this case, @cuducos ? I am unsure what to do when the content-length of the head request is zero.

Reply:
Yes, we have. E.g.:

```console
$  curl -I https://www.tesourotransparente.gov.br/ckan/dataset/abb968cb-3710-4f85-89cf-875c91b9c7f6/resource/eebb3bc6-9eea-4496-8bcf-304f33155282/download/TABMUN.CSV
HTTP/1.1 200 OK
Date: Mon, 14 Nov 2022 19:46:38 GMT
Server: Apache
Pragma: no-cache
Cache-Control: no-cache
Accept-Ranges: bytes
ETag: "1640107037.73-435942"
Last-Modified: Tue, 21 Dec 2021 17:17:17 GMT
Content-Range: bytes 0-435941/435942
Content-Type: text/csv
Via: 1.1 www.tesourotransparente.gov.br
Access-Control-Allow-Origin: https://www.tesouro.fazenda.gov.br/
```

In this case, we might try to:
* Get the size from the `Content-Range` response header
* I think it should fail the download if the server does not accept content range (extreme case, but might be a stretch goal)

Reply:
> 've searched for the implementation of Content-Range on the original HTTP lib and its not implemented as a default field

It's just a header of the response, then `resp.Header.Get("Content-Range")`.